app/src/main/java/ch/hikemate/app/model/route/ListOfHikeRoutesViewModel.kt

```diff
@@ -51,6 +51,26 @@ open class ListOfHikeRoutesViewModel(
   private suspend fun getRoutesAsync(onSuccess: () -> Unit = {}, onFailure: () -> Unit = {}) {
     withContext(dispatcher) {
       val area = area_.value ?: return@withContext
+
+      // Check if the area is on the date line
+      if (area.lonEast < area.lonWest) {
+        val bounds1 = BoundingBox(area.latNorth, 180.0, area.latSouth, area.lonWest)
+        val bounds2 = BoundingBox(area.latNorth, area.lonEast, area.latSouth, -180.0)
+        hikeRoutesRepository.getRoutes(
+            bounds = bounds1.toBounds(),
+            onSuccess = { routes1 ->
+              hikeRoutesRepository.getRoutes(
+                  bounds = bounds2.toBounds(),
+                  onSuccess = { routes2 ->
+                    hikeRoutes_.value = routes1 + routes2
+                    onSuccess()
+                  },
+                  onFailure = { _ -> onFailure() })
+            },
+            onFailure = { _ -> onFailure() })
```

Comment:
Question: is there any reason you did not do those two requests in parallel instead of sequentially? We might gain some time there, since both request will probably take time.

Reply:
How would you manage the callbacks in this case ? I'm not sure how to do it.

Reply:
Fair, it would make the code a little bit more complex and I can't tell you how to do it from the top of my head now, so I guess we can call it a day and fix this later if we notice strong problems with this additional delay.